dfastmi/batch/ConfigurationCheckerLegacy.py

```diff
@@ -0,0 +1,426 @@
+        celerity_hg = reach.proprate_high
+        celerity_lw = reach.proprate_low
+
+        (
+            q_threshold,
+            Q1,
+            apply_q1,
+            tstag,
+            fractions_of_the_year,
+            rsigma,
+        ) = self._batch_get_discharges(
+            reach, config, q_stagnant, celerity_hg, celerity_lw, nwidth
+        )
+        discharges = Q1
+        apply_q = apply_q1
+        time_mi = tuple(0 if discharges[i] is None or discharges[i]<=q_stagnant else fractions_of_the_year[i] for i in range(len(fractions_of_the_year)))
+        celerity = (celerity_lw, celerity_hg, celerity_hg)
+
+        return (discharges, apply_q, q_threshold, time_mi, tstag, fractions_of_the_year, rsigma, celerity)
+
+    def _batch_get_discharges(
+        self,
+        reach : ReachLegacy,
+        config: configparser.ConfigParser,
+        q_stagnant: float,
+        celerity_hg: float,
+        celerity_lw: float,
+        nwidth: float,
+    ) -> Tuple[
+        Optional[float],
+        QRuns,
+        Tuple[bool, bool, bool],
+        float,
+        Vector,
+        Vector,
+    ]:
+        """
+        Get the simulation discharges in batch mode (no user interaction).
+
+        Arguments
+        ---------
+        rivers : RiversObject
+            A dictionary containing the river data.
+        ibranch : int
+            Number of selected branch.
+        ireach : int
+            Number of selected reach.
+        config : configparser.ConfigParser
+            Configuration of the analysis to be run.
+        q_stagnant : float
+            Discharge below which the river flow is negligible [m3/s].
+        celerity_hg : float
+            Bed celerity during transitional and flood periods (from rivers configuration file) [m/s].
+        celerity_lw : float
+            Bed celerity during low flow period (from rivers configuration file) [m/s].
+        nwidth : float
+            Normal river width (from rivers configuration file) [m].
+
+        Results
+        -------
+        q_threshold : Optional[float]
+            River discharge at which the measure becomes active [m3/s].
+        three_characteristic_discharges : QRuns
+            Tuple of (at most) three characteristic discharges [m3/s].
+        apply_q : Tuple[bool, bool, bool]
+            A list of 3 flags indicating whether each value should be used or not.
+            The Q1 value can't be set to None because it's needed for char_times.
+        tstag : float
+            Fraction of year during which flow velocity is considered negligible [-].
+        fractions_of_the_year : Vector
+            A vector of values each representing the fraction of the year during which the discharge is given by the corresponding entry in Q [-].
+        rsigma : Vector
+            A vector of values each representing the relaxation factor for the period given by the corresponding entry in Q [-].
+        """
+        q_fit = reach.qfit
```

Comment:
is there a specific reason these are put in seperate variables and then together with reach are passed into the next method?